--- config_template.yml	2023-11-14 15:24:06.829665545 +0200
+++ config_template.yml.PAID	2023-11-14 15:25:19.790531113 +0200
@@ -157,20 +157,20 @@
 j11_par_executor: &j11_par_executor
   executor:
     name: java11-executor
-    #exec_resource_class: xlarge
-  parallelism: 4
+    exec_resource_class: medium
+  parallelism: 25
 
 j11_small_par_executor: &j11_small_par_executor
   executor:
     name: java11-executor
-    #exec_resource_class: xlarge
-  parallelism: 1
+    exec_resource_class: medium
+  parallelism: 10
 
 j11_small_parallel_executor: &j11_small_parallel_executor
   executor:
     name: java11-executor
-    #exec_resource_class: xlarge
-  parallelism: 1
+    exec_resource_class: medium
+  parallelism: 4
 
 j11_small_executor: &j11_small_executor
   executor:
@@ -178,30 +178,41 @@
     exec_resource_class: medium
   parallelism: 1
 
+j11_large_par_executor: &j11_large_par_executor
+  executor:
+    name: java11-executor
+    exec_resource_class: large
+  parallelism: 50
+
+j11_very_large_par_executor: &j11_very_large_par_executor
+  executor:
+    name: java11-executor
+    exec_resource_class: large
+  parallelism: 100
+
 j11_medium_par_executor: &j11_medium_par_executor
   executor:
     name: java11-executor
-    #exec_resource_class: xlarge
-  parallelism: 1
+    exec_resource_class: xlarge
+  parallelism: 4
 
 j11_seq_executor: &j11_seq_executor
   executor:
     name: java11-executor
-    #exec_resource_class: xlarge
     exec_resource_class: medium
   parallelism: 1 # sequential, single container tests: no parallelism benefits
 
 j17_par_executor: &j17_par_executor
   executor:
     name: java17-executor
-    #exec_resource_class: xlarge
-  parallelism: 4
+    exec_resource_class: medium
+  parallelism: 25
 
 j17_small_par_executor: &j17_small_par_executor
   executor:
     name: java17-executor
-    #exec_resource_class: xlarge
-  parallelism: 1
+    exec_resource_class: medium
+  parallelism: 10
 
 j17_small_executor: &j17_small_executor
   executor:
@@ -212,44 +223,56 @@
 j17_medium_par_executor: &j17_medium_par_executor
   executor:
     name: java17-executor
-    #exec_resource_class: xlarge
-  parallelism: 1
+    exec_resource_class: xlarge
+  parallelism: 4
+
+j17_large_par_executor: &j17_large_par_executor
+  executor:
+    name: java17-executor
+    exec_resource_class: large
+  parallelism: 50
 
 j17_seq_executor: &j17_seq_executor
   executor:
     name: java17-executor
-    #exec_resource_class: xlarge
+    exec_resource_class: medium
   parallelism: 1 # sequential, single container tests: no parallelism benefits
 
 j11_repeated_jvm_upgrade_dtest_executor: &j11_repeated_jvm_upgrade_dtest_executor
   executor:
     name: java11-executor
-  parallelism: 4
+    exec_resource_class: medium
+  parallelism: 25
 
 j11_repeated_utest_executor: &j11_repeated_utest_executor
   executor:
     name: java11-executor
-  parallelism: 4
+    exec_resource_class: medium
+  parallelism: 25
 
 j11_repeated_dtest_executor: &j11_repeated_dtest_executor
   executor:
     name: java11-executor
-  parallelism: 4
+    exec_resource_class: large
+  parallelism: 25
 
 j17_repeated_utest_executor: &j17_repeated_utest_executor
   executor:
     name: java17-executor
-  parallelism: 4
+    exec_resource_class: medium
+  parallelism: 25
 
 j17_repeated_dtest_executor: &j17_repeated_dtest_executor
   executor:
     name: java17-executor
-  parallelism: 4
+    exec_resource_class: large
+  parallelism: 25
 
 j11_repeated_upgrade_dtest_executor: &j11_repeated_upgrade_dtest_executor
   executor:
     name: java11-executor
-  parallelism: 4
+    exec_resource_class: xlarge
+  parallelism: 25
 
 j11_separate_jobs: &j11_separate_jobs
   jobs:
@@ -1976,7 +1999,7 @@
           target: testclasslist-system-keyspace-directory
 
   j11_dtests_vnode:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -1990,7 +2013,7 @@
           pytest_extra_args: '--use-vnodes --num-tokens=16 --skip-resource-intensive-tests'
 
   j11_dtests_offheap:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2004,7 +2027,7 @@
           pytest_extra_args: '--use-vnodes --num-tokens=16 --use-off-heap-memtables --skip-resource-intensive-tests'
 
   j17_dtests_vnode:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
     - attach_workspace:
         at: /home/cassandra
@@ -2019,7 +2042,7 @@
         pytest_extra_args: '--use-vnodes --num-tokens=16 --skip-resource-intensive-tests'
 
   j17_dtests_offheap:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2034,7 +2057,7 @@
           pytest_extra_args: '--use-vnodes --num-tokens=16 --use-off-heap-memtables --skip-resource-intensive-tests'
 
   j11_dtests:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2048,7 +2071,7 @@
           pytest_extra_args: '--skip-resource-intensive-tests'
 
   j17_dtests:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
     - attach_workspace:
         at: /home/cassandra
@@ -2063,7 +2086,7 @@
         pytest_extra_args: '--skip-resource-intensive-tests'
 
   j11_upgrade_dtests:
-    <<: *j11_par_executor
+    <<: *j11_very_large_par_executor
     steps:
     - attach_workspace:
         at: /home/cassandra
@@ -2077,7 +2100,7 @@
         pytest_extra_args: '--execute-upgrade-tests-only --upgrade-target-version-only --upgrade-version-selection all'
 
   j11_cqlsh_dtests_py3_vnode:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2092,7 +2115,7 @@
           extra_env_args: 'CQLSH_PYTHON=/usr/bin/python3.7'
 
   j11_cqlsh_dtests_py3_offheap:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2107,7 +2130,7 @@
           extra_env_args: 'CQLSH_PYTHON=/usr/bin/python3.7'
 
   j11_cqlsh_dtests_py38_vnode:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2125,7 +2148,7 @@
           python_version: '3.8'
 
   j11_cqlsh_dtests_py311_vnode:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2143,7 +2166,7 @@
           python_version: '3.11'
 
   j11_cqlsh_dtests_py38_offheap:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2161,7 +2184,7 @@
           python_version: '3.8'
 
   j11_cqlsh_dtests_py311_offheap:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2179,7 +2202,7 @@
           python_version: '3.11'
 
   j11_cqlsh_dtests_py3:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2194,7 +2217,7 @@
           extra_env_args: 'CQLSH_PYTHON=/usr/bin/python3.7'
 
   j11_cqlsh_dtests_py38:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2212,7 +2235,7 @@
           python_version: '3.8'
 
   j11_cqlsh_dtests_py311:
-    <<: *j11_par_executor
+    <<: *j11_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2230,7 +2253,7 @@
           python_version: '3.11'
 
   j17_cqlsh_dtests_py3_vnode:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2245,7 +2268,7 @@
           extra_env_args: 'CQLSH_PYTHON=/usr/bin/python3.7'
 
   j17_cqlsh_dtests_py3_offheap:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2260,7 +2283,7 @@
           extra_env_args: 'CQLSH_PYTHON=/usr/bin/python3.7'
 
   j17_cqlsh_dtests_py38_vnode:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2278,7 +2301,7 @@
           python_version: '3.8'
 
   j17_cqlsh_dtests_py311_vnode:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2296,7 +2319,7 @@
           python_version: '3.11'
 
   j17_cqlsh_dtests_py38_offheap:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2314,7 +2337,7 @@
           python_version: '3.8'
 
   j17_cqlsh_dtests_py311_offheap:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2332,7 +2355,7 @@
           python_version: '3.11'
 
   j17_cqlsh_dtests_py3:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2347,7 +2370,7 @@
           extra_env_args: 'CQLSH_PYTHON=/usr/bin/python3.7'
 
   j17_cqlsh_dtests_py38:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2365,7 +2388,7 @@
           python_version: '3.8'
 
   j17_cqlsh_dtests_py311:
-    <<: *j17_par_executor
+    <<: *j17_large_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2383,7 +2406,7 @@
           python_version: '3.11'
 
   j11_dtests_large_vnode:
-    <<: *j11_par_executor
+    <<: *j11_medium_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2397,7 +2420,7 @@
           pytest_extra_args: '--use-vnodes --num-tokens=16 --only-resource-intensive-tests --force-resource-intensive-tests'
 
   j11_dtests_large:
-    <<: *j11_par_executor
+    <<: *j11_medium_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2411,7 +2434,7 @@
           pytest_extra_args: '--only-resource-intensive-tests --force-resource-intensive-tests'
 
   j17_dtests_large_vnode:
-    <<: *j17_par_executor
+    <<: *j17_medium_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
@@ -2425,7 +2448,7 @@
           pytest_extra_args: '--use-vnodes --num-tokens=16 --only-resource-intensive-tests --force-resource-intensive-tests'
 
   j17_dtests_large:
-    <<: *j17_par_executor
+    <<: *j17_medium_par_executor
     steps:
       - attach_workspace:
           at: /home/cassandra
